src/PostgREST/Response.hs

```diff
@@ -292,9 +295,32 @@ optionalRollback AppConfig{..} ApiRequest{iPreferences=Preferences{..}} resp = d
 -- | Add headers not already included to allow the user to override them instead of duplicating them
 addHeadersIfNotIncluded :: [HTTP.Header] -> [HTTP.Header] -> [HTTP.Header]
 addHeadersIfNotIncluded newHeaders initialHeaders =
-  filter (\(nk, _) -> isNothing $ find (\(ik, _) -> ik == nk) initialHeaders) newHeaders ++
+  filter (\(nk, nv) -> isNothing $ find (\(ik, iv) -> ik == nk && nv == iv) initialHeaders) newHeaders ++
   initialHeaders
 
+{-
+  Usage: Filters out multiple Preference-Applied Headers from the list and concatenate them into a single Preference-Applied header
+  Example:
+
+  Input: [(hContentType, "application/json")
+         , (hPreferenceApplied, "tx=commit")
+         , (hPreferenceApplied, "return=minimal")]
+  Output: [(hContentType, "application/json"
+          , (hPreferenceApplied, "tx=commit, return=minimal")]
+
+-}
+concatPrefAppsHeaders :: [HTTP.Header] -> [HTTP.Header]
+concatPrefAppsHeaders headers = otherHeaders ++ [(HTTP.hPreferenceApplied, combinedPrefApps)]
+  where
+  prefApps = filter (\(k, _) -> k == HTTP.hPreferenceApplied) headers
+  otherHeaders = filter (\(k, _) -> k /= HTTP.hPreferenceApplied) headers
```

Comment:
Looks like we can use [partition](https://hackage.haskell.org/package/base-4.18.0.0/docs/Data-List.html#v:partition) here.